core/src/main/java/kafka/server/SharePartitionManager.java

```diff
@@ -631,4 +651,112 @@ public ShareFetchPartitionData(FetchParams fetchParams, String groupId, String m
             this.partitionMaxBytes = partitionMaxBytes;
         }
     }
+
+    static class ShareGroupMetrics {
+        /**
+         * share-acknowledgement (share-acknowledgement-rate and share-acknowledgement-count) - The total number of offsets acknowledged for share groups (requests to be ack).
+         * record-acknowledgement (record-acknowledgement-rate and record-acknowledgement-count) - The number of records acknowledged per acknowledgement type.
+         * partition-load-time (partition-load-time-avg and partition-load-time-max) - The time taken to load the share partitions.
+         */
+
+        public static final String METRICS_GROUP_NAME = "share-group-metrics";
+        public static final String PROTOCOL = "protocol";
```

Comment:
Hmm, I think we mistakenly put `protocol=share` for `share-group-metrics`. The `protocol=share` is more intended for `group-coordinator-metrics` where we need the differentiation.
cc: @AndrewJSchofield

Reply:
Indeed. That tag is only required for group coordinator metrics which apply to other kinds of groups also.

Reply:
@AndrewJSchofield 
Probably need to update kip description (JMX bean column)
```
kafka.server:type=share-group-coordinator-metrics,name=share-acknowledgement-rate,protocol=share

kafka.server:type=share-group-coordinator-metrics,name=share-acknowledgement-count,protocol=share 
```